### DISCUSSION TITLE
Any way to expose Git data?

### DISCUSSION THREAD

**Discussion by brycewray** [ElMassimo/iles#132] [Q&A] [answered]
@ElMassimo First of all, thank you for this amazing project! You seem to have resolved some issues that the Astro team, with all their many contributors, haven’t yet managed. :) [In fairness, of course, their project is trying to do quite a bit more; but on certain points that are important to me — notably, efficient and quick handling of Markdown and far better support for providing data to RSS and JSON feeds — yours is a better solution.]

**My question**: 

Does iles expose a repo’s Git data so it can be accessed within pages? Examples of what I mean are in these posts:

- (For Eleventy) https://www.aleksandrhovhannisyan.com/blog/eleventy-build-info/
- (For Hugo) https://www.brycewray.com/posts/2022/06/get-good-git-info-hugo/

Thanks in advance for any such info you may have time to provide!

**Comment by ElMassimo** [comment, marked as answer]
Hi Bryce! @brycewray Glad that you are enjoying îles!

There's no "official" approach, but it's something that can be achieved using [Modules](https://iles-docs.netlify.app/guide/plugins#modules-and-plugins), as in the case of the [documentation website](https://github.com/ElMassimo/iles/blob/9059b11e281aada9fbad70f961b33eb45123d10e/docs/iles.config.ts#L25).

Using the [`extendFrontmatter`](https://iles-docs.netlify.app/guide/plugins#islandspages) hook, you can [inject information from git](https://github.com/ElMassimo/iles/blob/7ac0a3bd70e982a5cfbb9ba29313385c5adb03b1/docs/modules/lastUpdated.ts#L6-L10) on each page. You can then [easily access that data](https://github.com/ElMassimo/iles/blob/9059b11e281aada9fbad70f961b33eb45123d10e/docs/src/components/LastUpdated.vue#L5) in the [page metadata](https://iles-docs.netlify.app/guide/development#using-page-data).

I've wanted to add a recipes section in the website covering common use cases such as these, but I haven't made time for that yet. Hopefully this gives you a good starting point to implement the use cases you have in mind 😃
